modyn/supervisor/internal/triggers/performancetrigger.py

```diff
@@ -0,0 +1,302 @@
+            )
+            if config.evaluation.label_transformer_function
+            else None
+        )
+
+    def init_trigger(self, context: TriggerContext) -> None:
+        self.context = context
+        self._init_dataloader_info()
+        self._init_model_downloader()
+
+    def inform(
+        self,
+        new_data: list[tuple[int, int, int]],
+        log: TriggerPolicyEvaluationLog | None = None,
+    ) -> Generator[int, None, None]:
+        # pylint: disable=too-many-locals
+        new_key_ts = self._leftover_data + [(key, timestamp) for key, timestamp, _ in new_data]
+        # reappending the leftover data to the new data requires incrementing the sample left until detection
+        self._sample_left_until_detection += len(self._leftover_data)
+
+        # index of the first unprocessed data point in the batch
+        processing_head_in_batch = 0
+
+        # Go through remaining data in new data in batches of `detect_interval`
+        while True:
+            if self._sample_left_until_detection - len(new_key_ts) > 0:
+                # No detection in this trigger because of too few data points to fill detection interval
+                self._leftover_data = new_key_ts
+                self._sample_left_until_detection -= len(new_key_ts)
+                return
+
+            # At least one detection, fill up window up to that detection
+            next_detection_interval = new_key_ts[: self._sample_left_until_detection]
+
+            # Update the remaining data
+            processing_head_in_batch += len(next_detection_interval)
+            new_key_ts = new_key_ts[len(next_detection_interval) :]
+
+            # Reset for next detection
+            self._sample_left_until_detection = self.config.detection_interval_data_points
+
+            # Run the evaluation (even if we don't use the result, e.g. for the first forced trigger)
+            self.data_density.inform_data(next_detection_interval)
+
+            num_samples, num_misclassifications, evaluation_scores = self._run_evaluation(
+                interval_data=next_detection_interval
+            )
+
+            self.performance_tracker.inform_evaluation(
+                num_samples=num_samples,
+                num_misclassifications=num_misclassifications,
+                evaluation_scores=evaluation_scores,
+            )
+
+            policy_decisions: dict[str, bool] = {}
+
+            # The first ever detection will always trigger
+            if not self._triggered_once:
+                # If we've never triggered before, always trigger
+                self._triggered_once = True
+                triggered = True
+
+                # in the first interval we don't evaluate the decision policies as they might require one trigger
+                # to have happened before in order to derive forecasts
+
+            else:
+                # evaluate the decision policies
+                any_policy_triggered = False
```

Comment:
nit: i think you can just use triggered and say triggered |= policy_decisions[policy_name]